### PR TITLE
feat: use fade-out effect on the filtered regions in the matrix

### PR DESCRIPTION
**Pull request by sehilyi**
This uses the consistent 'fade-out' effect on the filtered-out items in the context view.

Screen Shot 2022-08-12 at 19 45 02

To draw a semi-transparent overlay, I basically separated this fade-out region into multiple rectangles. The following demonstrates how the white regions are actually drawn (stroke colors of individual rectangles being `red`):

Screen Shot 2022-08-12 at 19 49 17

Because we are not able to use data transforms (since Gosling uses HiGlass to render matrix, and HiGlass does not support filtering), I was not able to find an easier approach to implement the 'fade-out effect.' If data transforms are available, we could just, for example, overlay (1) a regular matrix, (2) a big semi-transparent white rectangle that covers the entire region of the matrix, and (3) a filtered matrix that shows only the selected samples.

### CODE REVIEW

**Comment by sehilyi** [comment]
To draw a semi-transparent overlay, I basically separated this fade-out region into multiple rectangles. The following demonstrates how the white regions are actually drawn (stroke colors of individual rectangles being `red`):

Screen Shot 2022-08-12 at 19 49 17

File: front/src/components/Gosling.tsx [hunk 1]
```diff
@@ -44,6 +44,56 @@ export default class GoslingVis extends React.Component<Props, {}> {
...
+    const sorted = labelJSON.sort((a, b) => a.start - b.start);
+    sorted.forEach(({ start, end }, i) => {
+      const lb: [number, number] = [start, end]; // the left bottom point of a rectangle
+      if(outerPoints.length === 0 || end > outerPoints[outerPoints.length - 1][1]) {
+        // This means `lb` is an outer point
+        outerPoints.push(lb);
+      }
+    });
+    const overlayRects: Record<string, string | number>[] = [];
+    const template = { chromosome: 'chr5' }; // Support other chromosomes as well?
```
Comment (sehilyi) [inline front/src/components/Gosling.tsx:58]: Would it be needed to fade out entire chromosomes other than chromosome 5? I currently only showed the semi-transparent overlay to chromosome 5 as it looked like only chromosome 5 contained any data items.

Comment (wangqianwen0418) [reply front/src/components/Gosling.tsx:58]: I only used chromosome 5 for the Hi-C dataset so only fading out chr5 should be fine.

Comment (wangqianwen0418) [reply front/src/components/Gosling.tsx:58]: Do we add a data transform to only show the CHR 5 data?

Comment (sehilyi) [reply front/src/components/Gosling.tsx:58]: The Matrix is rendered directly by HiGlass which does not allow us to apply data transforms.

Maybe we can additionally overlay semi-transparent or opaque white rectangles on all chromosomes?

Comment (wangqianwen0418) [reply front/src/components/Gosling.tsx:58]: Yes, you mentioned that before. Let's fade out other CHRs and this branch is ready to merge.
Thanks a lot for creating this PR!

**Review by wangqianwen0418 — APPROVED** [review]

**Comment by sehilyi** [comment]
Some screenshots after making a further change:

Screen Shot 2022-08-14 at 10 31 03


Screen Shot 2022-08-14 at 10 37 58

**Pull request merged**